### ISSUE TITLE
`assert x1 == x2 or y1 == y2` exception

### ISSUE THREAD

**Issue by memegauste** [Kozea/WeasyPrint#2223]
Affected versions: _62.0_, _62.1_, _62.2_, _62.3_

Description:
Exception occurs when using `django-weasyprint` package. I also tested versions of weasyprint prior to 62 release, they are not working anyway with `django-weasyprint` (different error occurs probably because it uses different code).

**HTML code that raises exception:**
https://pastebin.com/2xHH1Lw9

**Custom defined view:**
```
class PrintCallHistoryView(WeasyTemplateResponseMixin, DetailView):  # noqa: D106

    pdf_attachment = False
    template_name = 'pdf/history.html'
    response_class = BasePDFTemplateResponse
    object_type = ''
    object = None
    access_groups = ['Operator', 'Caller']

    def get_object(self):
        """Get the object and object_type based on the UUID."""
        self.object_type, self.object = Caller.objects.get_by_uuid(
            uuid=self.kwargs.get('uuid'), with_type=True
        )
        return self.object

    def get_pdf_filename(self):
        now = timezone.now().strftime('%Y%m%d-%H%M')
        return f'call_history-{now}.pdf'
```

**Requirements:**
```
django-weasyprint==2.3.0
weasyprint==62.1
```

**Exception screenshots:**

![image](https://github.com/user-attachments/assets/438bc0ac-7ffc-48a6-85a1-c6298607fa36)

![image](https://github.com/user-attachments/assets/317c4605-09be-4ed8-b9e2-f990ed29ac83)

**Comment by PSzczepanski1996** [comment]
Okey, so I managed to catch the problem. It occurs due to `custom.min.css` inclusion in the `base.html` or either `View` class in Django Weasyprint. Due to company policy I can paste the whole file here, but it occurs probably due to `flex` table implementation.

**Comment by liZe** [comment]
Hi,

Could you please share the generated HTML and the CSS, instead of the Django template? Without that, we can’t reproduce your problem.

**Comment by liZe** [comment]
I really don’t know how this assertion could be false.

https://github.com/Kozea/WeasyPrint/blob/56acbc422e3aeb7974ae8c4d45e9726cfaf10bea/weasyprint/draw.py#L1131-L1133

https://github.com/Kozea/WeasyPrint/blob/56acbc422e3aeb7974ae8c4d45e9726cfaf10bea/weasyprint/draw.py#L1154-L1156

Width is set to 0 for vertical lines, and height is set to 0 for horizontal lines. It means that `x1 == x2`, or that `y1 == y2`.

**Comment by liZe** [comment]
Don’t hesitate to add a comment if you can share your HTML and CSS files.

**Comment by elazarcoh** [comment]
@liZe 
I have the same asserting error. For me it is the case that `y1` and `y2` both are `nan` (don't know why), but in python `nan` are never equal.

**Comment by liZe** [comment]
> I have the same asserting error. For me it is the case that `y1` and `y2` both are `nan` (don't know why), but in python `nan` are never equal.

Makes sense. Could you please share a HTML/CSS sample to reproduce the problem?

**Comment by elazarcoh** [comment]
Sorry, I already have worked around it by using different css, and can't reproduce it anymore. I think it was something to do with tables and flex column, but I'm not sure exactly what was the fix so I can't reverse it.
As far as I'm concerned, this issue is no block for me. 

If someone else have the same issue, please fill here an example.

**Comment by liZe** [comment]
> I think it was something to do with tables and flex column, but I'm not sure exactly what was the fix so I can't reverse it.

Then it’s probably a problem with flex, that will go away when flex is rewritten (#2362).

> If someone else have the same issue, please fill here an example.

That would be really helpful!

**Comment by liZe** [comment]
Closing, as flex has been rewritten and no example is available to reproduce. If anyone hits this error with the current main branch, please add a comment with a sample attached.
